Make keystore-password, keystore, trustore keys in secret configurable

studypyth opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

**What this PR does**:
Make keystore-password, keystore, trustore keys in secret configurable

**Which issue(s) this PR fixes**:
Fixes #718 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

adejanovski commented on pkg/encryption/encryption_stores.go:
```diff
@@ -6,15 +6,25 @@ import (
 type Stores struct {
-	// ref to the secret that contains the keystore and its password
-	// the expected format of the secret is a "keystore" entry and a "keystore-password" entry
+	// ref to the secret that contains the keystore and its can contain password if not used keystorePasswordSecretRef
+	// if not specified key explicitly "keystore" entry and a "keystore-password" entry will be used
```
```suggestion
	// ref to the secret that contains the keystore and optionally its password (which can also be specified through 
the keystorePasswordSecretRef field)
	// if keys are not specified, "keystore" entry and a "keystore-password" entry will be used
```

studypyth replied:
Fixed

adejanovski commented on pkg/encryption/encryption_stores.go:
```diff
@@ -6,15 +6,25 @@ import (
 
-	// ref to the secret that contains the truststore and its password
-	// the expected format of the secret is a "truststore" entry and a "truststore-password" entry
+	// ref to the secret that contains the truststore and can its contain password if not used truststoreSecretRef
+	// if not specified key explicitly "keystore" entry and a "keystore-password" entry will be used
```
```suggestion
	// ref to the secret that contains the truststore and optionally its password (which can also be specified through the truststorePasswordSecretRef field)
	// if keys are not specified explicitly, "keystore" entry and a "keystore-password" entry will be used
```

studypyth replied:
Fixed

adejanovski reviewed (changes requested):
Hi @studypyth,

thanks a lot for pushing this PR!
I love the fact that you managed to retain backwards compatibility while allowing to use separate secrets and keys for the stores and their passwords.
I have a small request for you about variable name changes, and if you wouldn't mind it would be great to update the docs [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/docs/encryption/README.md) to explain the new capabilities. 

Thanks again

studypyth commented:
Hi @adejanovski, thanks for the review, I've updated the docs, fixed naming issues and updated CRD and autogenerated files from controller-gen(zz_generated.deepcopy.go)

studypyth commented on pkg/cassandra/encryption.go:
```diff
@@ -204,13 +203,39 @@ func ReadEncryptionStoresSecrets(ctx context.Context, klusterKey types.Namespace
+	switch storeName {
+	case encryption.StoreNameKeystore:
+		general = stores.KeystoreSecretRef
+		specific = stores.KeystorePasswordRef
```
good point, i have renamed variables
